src/widget/wtrackmenu.cpp

```diff
@@ -1595,6 +1625,161 @@ void WTrackMenu::slotClearAllMetadata() {
             &trackOperator);
 }
 
+namespace {
+
+class RemoveTrackFilesFromDiskTrackPointerOperation : public mixxx::TrackPointerOperation {
+  public:
+    mutable QList<TrackRef> tr_tracksToPurge;
+    mutable QList<QString> tr_tracksToKeep;
+
+  private:
+    void doApply(
+            const TrackPointer& pTrack) const override {
+        auto trackRef = TrackRef::fromFileInfo(
+                pTrack->getFileInfo(),
+                pTrack->getId());
+        VERIFY_OR_DEBUG_ASSERT(trackRef.isValid()) {
+            return;
+        }
+        QString location = pTrack->getLocation();
+        QFile file(location);
+        if (file.exists() && !file.remove()) {
+            // Deletion failed, log warning and queue location for the
+            // Failed Deletions warning.
+            qWarning()
+                    << "Queued file"
+                    << location
+                    << "could not be deleted. Track is not purged";
+            tr_tracksToKeep.append(location);
+            return;
```

Comment:
This `return`m statement is both redundant and inconsistent with the other if arm.

Reply:
:+1: